src/components/ec-btn/ec-btn.spec.js

```diff
@@ -0,0 +1,200 @@
+      const wrapper = mountBtn({
+        disabled: true,
+      });
+
+      expect(wrapper.attributes().disabled).toBe('disabled');
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it('should render a button with an icon when the icon prop is defined', () => {
+      const wrapper = mountBtn({
+        icon: 'simple-check',
+      });
+
+      expect(wrapper.find('.ec-btn__icon').exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it('should render a button with an outline when "outline" prop is set to true', () => {
+      const wrapper = mountBtn({
+        outline: true,
+      });
+
+      expect(wrapper.find('.ec-btn--outline').exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it('should render a full width button when "fullWidth" prop is set to true', () => {
+      const wrapper = mountBtn({
+        fullWidth: true,
+      });
+
+      expect(wrapper.find('.ec-btn--full-width').exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it.each(['primary', 'secondary', 'success', 'error', 'warning'])('should render a button with type "%s" when type prop is set', (type) => {
+      const wrapper = mountBtn({
+        type,
+      });
+
+      expect(wrapper.find(`.ec-btn--${type}`).exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it('should render a reverse button when reverse is set to true', () => {
+      const wrapper = mountBtn({
+        reverse: true,
+      });
+
+      expect(wrapper.find('.ec-btn--reverse').exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it('should render a button with a spinner loader if loading is set to true', () => {
+      const wrapper = mountBtn({
+        loading: true,
+      });
+
+      expect(wrapper.attributes().disabled).toBe('disabled');
+      expect(wrapper.find('.ec-btn__spinner').exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it('should render a button with type submit if submit is set to true', () => {
+      const wrapper = mountBtn({
+        submit: true,
+      });
+
+      expect(wrapper.attributes().type).toBe('submit');
+      expect(wrapper.element).toMatchSnapshot();
+    });
+  });
+
+  describe(':slots', () => {
+    it('should render a button with "Click me!" text', () => {
+      const wrapper = mountBtn(
+        {},
+        {
+          slots: {
+            default: 'Click Me!',
+          },
+        },
+      );
+      expect(wrapper.find('.ec-btn__spinner').exists()).toBe(false);
+      expect(wrapper.findByDataTest('ec-btn__text-loader-slot').exists()).toBe(false);
+      expect(wrapper.findByDataTest('ec-btn__default-slot').exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it('should render a loading button with a spinner loader and a "Click me!" text set to invisble', () => {
```

Comment:
We cannot check something is invisible in unit tests. We don't have the CSS.

Reply:
I kept the test but rephrased the description. I would like to test if the default slot has the loading class (so should be invisible unless someone amends the class)